database/deltas/execute.go

```diff
@@ -53,16 +55,23 @@ func Execute(config database.InstanceConfig, schema string, deltas []*Delta) err
 	config.Connection = setMultiStatement(config.Connection)
 	db := database.Initialize(&config)
 
+	getLock := func() error {
+		locked, err := database.AcquireDatabaseLock(db, LockName, 0)
+		if nil != err {
+			return err
+		}
+		if !locked {
+			return errors.New("unable to acquire lock")
+		}
+		return nil
+	}
+
 	log.Infof("executing %d deltas on %s", len(deltas), schema)
-	// we want to just bail if someone else has the lock
-	locked, err := database.AcquireDatabaseLock(db, LockName, 0)
+	err = net.BackoffExtended(getLock, 10, time.Second, 10*time.Second)
```

Comment:
How would you like the timing components passed in?

Reply:
Let's put this on the instance config (`gadget/database/model.go (56)`)